### PR TITLE
Blink Detector Plugin

### PR DESCRIPTION
**Pull request by papr**
Mergable version from #581 + PEP 8 changes + code cleanup

### CODE REVIEW

File: pupil_src/capture/blink_detection.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+class Blink_Detection(Plugin):
+    """
+    DisplayGaze shows the three most
+    recent gaze position on the screen
+    """
+
+    def __init__(self, g_pool):
+        super(Blink_Detection, self).__init__(g_pool)
+        self.order = .8
+        self.history_length = 21 # Should be odd
```
Comment (papr) [inline pupil_src/capture/blink_detection.py]: What is the exact reason for this number to be odd?

File: pupil_src/capture/blink_detection.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            if self.is_blink and act < -2:
+                self.is_blink = False
+
+            print str(self.is_blink) + " " + str(act)
+
+            blink_entry = {
+                'topic': 'blink',
+                'activation': act,
+                'timestamp': self.timestamp_histories[pt['id']][10],
+                'is_blink': int(self.is_blink)
```
Comment (papr) [inline pupil_src/capture/blink_detection.py]: What is the reason for `is_blink` to be an `int`? What does speak against it staying a `bool`?

File: pupil_src/capture/blink_detection.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                self.is_blink = True
+            if self.is_blink and act < -2:
+                self.is_blink = False
+
+            print str(self.is_blink) + " " + str(act)
+
+            blink_entry = {
+                'topic': 'blink',
+                'activation': act,
+                'timestamp': self.timestamp_histories[pt['id']][10],
```
Comment (papr) [inline pupil_src/capture/blink_detection.py]: `10` can be replaced by `self. history_length // 2`, correct?

File: pupil_src/capture/blink_detection.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            if not self.is_blink and act > 7:
+                logger.error("Bink")
+                self.is_blink = True
+            if self.is_blink and act < -2:
+                self.is_blink = False
+
+            print str(self.is_blink) + " " + str(act)
+
+            blink_entry = {
+                'topic': 'blink',
```
Comment (papr) [inline pupil_src/capture/blink_detection.py]: Is the `topic` entry necessary if the entry is added to `events['blinks']`? Are there other entries planned?

File: pupil_src/capture/blink_detection.py (hunk 2 above)
Comment (marc-tonsen) [inline pupil_src/capture/blink_detection.py]: I got pickle errors when I set that to a bool

File: pupil_src/capture/blink_detection.py (hunk 3 above)
Comment (marc-tonsen) [inline pupil_src/capture/blink_detection.py]: Correct. Ideally you would also set `self.history_length` to something that adapts to the framerate

**Comment by papr** [comment]
Ready for merge.

**Comment by marc-tonsen** [comment]
The plugin will only work well with the 3D detector, if the `confidence` value in events is changed to 2D pupil detection confidence

**Comment by mkassner** [comment]
@papr  can you make this mergable?

**Comment by papr** [comment]
@mkassner did you make the changes to confidence as @marc-tonsen requested?

**Comment by mkassner** [comment]
@papr  no not yet.

**Pull request merged**